src/engine/GroupBy.cpp

```diff
@@ -1489,10 +1491,21 @@ IdTable GroupBy::computeGroupByForHashMapOptimization(
     aggregationTimer.cont();
     for (auto& aggregateAlias : aggregateAliases) {
       for (auto& aggregate : aggregateAlias.aggregateInfo_) {
+        // The code below assumes that DISTINCT is not supported yet.
+        AD_CORRECTNESS_CHECK(aggregate.expr_->isAggregate() ==
+                             sparqlExpression::SparqlExpression::
+                                 AggregateStatus::NonDistinctAggregate);
         // Evaluate child expression on block
         auto exprChildren = aggregate.expr_->children();
+        // `COUNT(*)` is the only expression without children, so we fake the
+        // expression result in this case by providing an arbitrary, constant
+        // and defined value. This value will be verified as non-undefined by
+        // the `CountExpression` class and ignored afterward as long as
+        // `DISTINCT` is not set (which is not supported yet).
         sparqlExpression::ExpressionResult expressionResult =
-            exprChildren[0]->evaluate(&evaluationContext);
+            exprChildren.empty()
+                ? Id::makeFromBool(true)
+                : exprChildren[0]->evaluate(&evaluationContext);
```

Comment:
This code (including its lengthy explanation) is duplicated in the `LazyGroupBy.cpp`. please factor it out.
(`evaluateSingleChildIfNotCountStar` or something could be a name for the function).